lib/src/web_socket/browser/sockjs.dart

```diff
@@ -0,0 +1,109 @@
+// Copyright 2015 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+library w_transport.src.web_socket.browser.sockjs;
+
+import 'dart:async';
+import 'dart:html';
+import 'dart:typed_data';
+
+import 'package:sockjs_client/sockjs_client.dart' as sockjs;
+
+import 'package:w_transport/src/web_socket/common/w_socket.dart';
+import 'package:w_transport/src/web_socket/w_socket.dart';
+import 'package:w_transport/src/web_socket/w_socket_close_event.dart';
+import 'package:w_transport/src/web_socket/w_socket_exception.dart';
+
+class SockJSSocket extends CommonWSocket implements WSocket {
+  static Future<WSocket> connect(Uri uri,
+      {bool debug: false,
+      bool noCredentials: false,
+      List<String> protocolsWhitelist}) async {
+    if (uri.scheme == 'ws') {
+      uri = uri.replace(scheme: 'http');
+    } else if (uri.scheme == 'wss') {
+      uri = uri.replace(scheme: 'https');
+    }
+
+    sockjs.Client client = new sockjs.Client(uri.toString(),
+        debug: debug == true,
+        noCredentials: noCredentials == true,
+        protocolsWhitelist: protocolsWhitelist);
+
+    // Listen for and store the close event. This will determine whether or
+    // not the socket connected successfully, and will also be used later
+    // to handle the web socket closing.
+    var closed = client.onClose.first;
```

Comment:
The sockjs library is doing something weird here.. I tried typing it as `Future<CloseEvent>` with `CloseEvent` from `dart:html` and with `CloseEvent` from the sockjs library, but both throw in checked mode. It's like they're using the `_Future` class somehow